### PR TITLE
HS-973,HS-975: Externalize encryption key as kubernetes secret

### PR DESCRIPTION
**Pull request by thomasbigger584**
## Description
HS-973: Create and store encryption private key as kubernetes secret
HS-975: Use externalized secret in the Azure credentials encryption

Verified the injection of the environment variable by printing it to the console for test purpose only.

## Jira link(s)
- https://issues.opennms.org/browse/HS-973
- https://issues.opennms.org/browse/HS-975

## Checklist
* [ ] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.

### CODE REVIEW

**Review by jkapponms — APPROVED** [review]
lgtm

File: charts/opennms/templates/opennms/inventory/inventory-encryptionkey-secret.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Values.OpenNMS.Inventory.ServiceName }}-encryption-key
+  namespace: {{ .Release.Namespace }}
+data:
+  {{- $secretData := get (lookup "v1" "Secret" .Release.Namespace (print .Values.OpenNMS.Inventory.ServiceName "-encryption-key")) "data" | default dict }}
+  encryptionKey: "{{ (get $secretData "encryptionKey" | b64dec) | default .Values.OpenNMS.Inventory.EncryptionKey | default (randAscii 32) | b64enc }}"
```
Comment (geraldhumphries) [inline charts/opennms/templates/opennms/inventory/inventory-encryptionkey-secret.yaml:8]: Not a review comment: About `default .Values.OpenNMS.Inventory.EncryptionKey`, we might not want any keys or passwords in the Helm chart's values in the future. This is good right now.

File: charts/opennms/values.yaml [hunk 2]
```diff
@@ -112,6 +112,7 @@ OpenNMS:
       Requests:
         Cpu: 100m
         Memory: 250Mi
+    EncryptionKey: encryption-key
```
Comment (geraldhumphries) [inline charts/opennms/values.yaml:115]: We shouldn't have a default value here. Use `EncryptionKey: ~` to set it to null by default, which will make Helm generate it when it applies the Secret.

File: install-local/install-local-opennms-horizon-stream-custom-images-values.yaml [hunk 3]
```diff
@@ -72,6 +72,7 @@ OpenNMS:
       Requests:
         Cpu: "0"
         Memory: "0"
+    EncryptionKey: encryption-key
```
Comment (geraldhumphries) [inline install-local/install-local-opennms-horizon-stream-custom-images-values.yaml:75]: This values file is overlaid on top of the chart's values (`charts/opennms/values.yaml`). There's no need to add values here that match the default, only do that if you need to change something.

File: install-local/install-local-opennms-horizon-stream-values.yaml [hunk 4]
```diff
@@ -64,6 +64,7 @@ OpenNMS:
       Requests:
         Cpu: "0"
         Memory: "0"
+    EncryptionKey: encryption-key
```
Comment (geraldhumphries) [inline install-local/install-local-opennms-horizon-stream-values.yaml:67]: Same comment re: default values in Helm.

File: inventory/src/main/resources/application.yml [hunk 5]
```diff
@@ -44,6 +44,8 @@ keycloak:
   base-url: http://onms-keycloak:8080/auth
 
 inventory:
+  encryption:
+    key: encryption-key
```
Comment (geraldhumphries) [inline inventory/src/main/resources/application.yml:48]: Perhaps leave this empty. A default encryption key anywhere is probably a security risk.

**Review by geraldhumphries — CHANGES_REQUESTED** [review]

**Pull request merged**